osquery/tables/networking/darwin/wifi.mm

```diff
@@ -147,15 +146,23 @@ QueryData genKnownWifiNetworks(QueryContext& context) {
   }
 
   QueryData results;
-  auto count = CFDictionaryGetCount(networks);
-  const void* keys[count];
-  const void* values[count];
-  CFDictionaryGetKeysAndValues(networks, keys, values);
-
-  for (CFIndex i = 0; i < count; i++) {
-    parseNetworks((CFDictionaryRef)values[i], results);
+  if (CFGetTypeID(networks) == CFArrayGetTypeID()) {
+    auto count = CFArrayGetCount((CFArrayRef)networks);
+    for (CFIndex i = 0; i < count; i++) {
+      parseNetworks(
+          (CFDictionaryRef)CFArrayGetValueAtIndex((CFArrayRef)networks, i),
+          results);
+    }
+  } else if (CFGetTypeID(networks) == CFDictionaryGetTypeID()) {
+    auto count = CFDictionaryGetCount((CFDictionaryRef)networks);
+    const void* keys[count];
+    const void* values[count];
+    CFDictionaryGetKeysAndValues((CFDictionaryRef)networks, keys, values);
+
+    for (CFIndex i = 0; i < count; i++) {
+      parseNetworks((CFDictionaryRef)values[i], results);
```

Comment:
`parseNetworks` is already doing a `nullpr` check, I could add a check for `CFGetTypeID(network) != CFDictionaryGetTypeID()`